test/upgrade/service_preupgrade_test.go

```diff
@@ -35,7 +36,11 @@ func TestRunLatestServicePreUpgrade(t *testing.T) {
 	names.Service = serviceName
 	names.Image = test.PizzaPlanet1
 
-	resources, err := v1a1test.CreateRunLatestServiceLegacyReady(t, clients, &names, &v1a1test.Options{})
+	resources, err := v1a1test.CreateRunLatestServiceLegacyReady(t, clients, &names, &v1a1test.Options{
+		RevisionTemplateAnnotations: map[string]string{
```

Comment:
Do you mind using the variadic ServiceOptions instead of `v1a1test.Options`? The helper I think is `WithConfigAnnotations`. We want to drop the usage of the `Options` struct in favor of using functional Options.

Reply:
OK. But it would be good to get rid of the other usages from the other tests (not as part of this PR) so that it doesn't confuse more people.

Reply:
Sure, i think we're currently lazy updating them as we need to touch files.